### PR TITLE
Fix printing an empty PauliList

### LINKED ISSUE #7359: Error printing empty `PauliList`
### Environment

- **Qiskit Terra version**:
  tag v0.19.0
- **Python version**:
 3.9 
- **Operating system**: 
 Linux

### What is happening?

```python
import numpy as np
from qiskit.quantum_info import SparsePauliOp, PauliList

x = np.array([], dtype=bool)
z = np.array([], dtype=bool)

pauli_list = PauliList.from_symplectic(x, z)

# Exception is raised
try:
    print(pauli_list)
except ZeroDivisionError as err:
    print(f"Can't print pauli_list: {err}")

# No problem printing SparsePauliOp
print(SparsePauliOp(pauli_list))
```
prints
```
Can't print pauli_list: integer division or modulo by zero
SparsePauliOp([''],
              coeffs=[1.+0.j])
```

Part of the stack trace:
```
~/code/github/Qiskit/oneenv/qiskit-terra/qiskit/quantum_info/operators/symplectic/pauli_list.py in _truncated_str(self, show_class)
    194         stop = self._num_paulis
    195         if self.__truncate__:
--> 196             max_paulis = self.__truncate__ // self.num_qubits
    197             if self._num_paulis > max_paulis:
    198                 stop = max_paulis
```

### How can we reproduce the issue?

Enter the code above

### What should happen?

No error should be thrown.

### Any suggestions?

check before dividing by zero

### PR DESCRIPTION
**Pull request by amyels**
### Summary

Updated self.num_qubits to check if larger than 0 to avoid dividing by 0 

### Details and comments

Closes #7359

### CODE REVIEW

**Review by ikkoham — DISMISSED** [review]
Thank you very much. The change looks good. Could you add a test to pass empty PauliList to the `str()` function, and add release notes?

**Comment by ikkoham** [comment]
@amyels You are a first time contributor, so if you are not sure about these additions, please feel free to ask here.

**Comment by jlapeyre** [comment]
This looks great! I think it's ready for merge.

**Review by jlapeyre — DISMISSED** [review]

**Review by jlapeyre — CHANGES_REQUESTED** [review]
EDITED: CI is blocking the merge because a file needs to be formatted by `black`.

[Here are instructions](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md#pull-request-checklist) giving one way to reformat with `black`. (In item number 1). In general, we need to reformat PRs with `black` after all changes have been made.

Otherwise, this looks great! The changes requested above by @ikkoham have been made.

**Review by jlapeyre — APPROVED** [review]

File: releasenotes/notes/fix-empty-string-pauli-list-init-4d978fb0eaf1bc70.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+fixes:
+  - |
+    Fixed initialization of empty symplectic matrix in :meth:'PauliList.from_symplectic' in :class:'.PauliList' class
+    For example::
+      from qiskit.quantum_info.operators import PauliList
+
+      x = np.array([], dtype=bool).reshape((1,0))
+      z = np.array([], dtype=bool).reshape((1,0))
+      pauli_list = PauliList.from_symplectic(x, z)
```
Comment (mtreinish) [inline releasenotes/notes/fix-empty-string-pauli-list-init-4d978fb0eaf1bc70.yaml:10]: Try this:

```suggestion
    Fixed initialization of empty symplectic matrix in :meth:'PauliList.from_symplectic' in :class:'.PauliList' class
    For example::

        from qiskit.quantum_info.operators import PauliList

        x = np.array([], dtype=bool).reshape((1,0))
        z = np.array([], dtype=bool).reshape((1,0))
        pauli_list = PauliList.from_symplectic(x, z)
```

**Review by mtreinish — APPROVED** [review]

**Pull request merged**